octodns/provider/ns1.py

```diff
@@ -1020,17 +1128,52 @@ def _params_for_SRV(self, record):
                   for v in record.values]
         return {'answers': values, 'ttl': record.ttl}, None
 
+    def _get_ns1_filters(self, ns1_zone_name):
+        ns1_filters = {}
+        ns1_zone = self._client.zones_retrieve(ns1_zone_name)
+        for ns1_record in ns1_zone['records']:
+            if ns1_record.get('tier', 1) > 1:
+                # Need to get the full record data for geo records
+                full_rec = self._client.records_retrieve(ns1_zone_name,
+                                                         ns1_record['domain'],
+                                                         ns1_record['type'])
+                if 'filters' in full_rec:
+                    filter_key = ns1_record['domain'] + '.'
```

Comment:
octoDNS's preferred style is to use format over concatentation.

```suggestion
                    filter_key = '{}.'format(ns1_record['domain'])
```